pulsar-common/pom.xml

```diff
@@ -131,6 +131,11 @@
       <artifactId>netty-tcnative-boringssl-static</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>io.netty.incubator</groupId>
+      <artifactId>netty-incubator-transport-classes-io_uring</artifactId>
```

Comment:
Technically, this dependency is transitive from `netty-incubator-transport-native-io_uring`. In a shaded env we should have this dependency in the classpath. I don't have a deep investigation how it fails. In my local env, even without this patch, everything works fine.

Reply:
It's already shaded in because the shaded modules have the wildcard rule with `io.netty.incubator` groupId.

Are you on linux ? because on Linux the native libs are loaded. On MacOs is easy to reproduce

Reply:
I've tested on macOS 12.6 Apple Silicon M1.

Reply:
Maybe it depends on your maven version